### PR TITLE
DLP samples

### PR DESCRIPTION
**Pull request by jabubake**
Pending DLP library release to Maven

### CODE REVIEW

File: dlp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    </dependencies>
+  </dependencyManagement>
+  <!--- End of workaround -->
+
+  <dependencies>
+    <!-- [START dlp_maven] -->
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-dlp</artifactId>
+      <!-- TODO update with release version -->
```
Comment (jabubake) [inline dlp/pom.xml]: Note : this needs to be updated to a release version prior to merge.

File: dlp/src/main/java/com/example/dlp/Redact.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+              ReplaceConfig.newBuilder()
+                  .setInfoType(infoType)
+                  .setReplaceWith(replacement)
+                  .build());
+        }
+      }
+
+      RedactContentResponse contentResponse = dlpClient.redactContent(
+          inspectConfig, Collections.singletonList(contentItem), replaceConfigs);
+      if (contentResponse.getItemsCount() > 0) {
```
Comment (lesv) [inline dlp/src/main/java/com/example/dlp/Redact.java]: Is this needed?

File: dlp/src/test/java/com/example/dlp/RedactIT.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+    assertNotNull(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+  }
+
+  @Test
+  public void testInfoTypesInStringAreReplaced() throws Exception {
+    String text =
+        "\"My phone number is (2234) 456-7890 and my email address is gary@somedomain.com\"";
```
Comment (lesv) [inline dlp/src/test/java/com/example/dlp/RedactIT.java]: one two many?

File: dlp/src/test/java/com/example/dlp/InspectIT.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  private PrintStream out;
+  private String bucket;
+  private String datastoreKind;
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+    assertNotNull(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
```
Comment (jabubake) [inline dlp/src/test/java/com/example/dlp/InspectIT.java]: do you mean the assertion ?  default auth did n't work: https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2231

File: dlp/src/test/java/com/example/dlp/InspectIT.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    assertNotNull(System.getenv("DLP_BUCKET_ID"));
```
Comment (jabubake) [inline dlp/src/test/java/com/example/dlp/InspectIT.java]: I added notes in the README.md on defining these. Do you have a better approach in mind ?

File: dlp/src/main/java/com/example/dlp/Redact.java (hunk 2 above)
Comment (jabubake) [inline dlp/src/main/java/com/example/dlp/Redact.java]: Removed.

File: dlp/src/test/java/com/example/dlp/RedactIT.java (hunk 3 above)
Comment (jabubake) [inline dlp/src/test/java/com/example/dlp/RedactIT.java]: Removed.

**Comment by jabubake** [comment]
@lesv PTAL when you get a chance.

**Comment by jabubake** [comment]
@ace-n fyi

File: dlp/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+## Integration tests
+### Setup
+- [Create a Google Cloud Storage bucket](https://console.cloud.google.com/storage) and upload [test.txt](src/test/resources/test.txt).
+- [Create a Google Cloud Datastore](https://console.cloud.google.com/datastore) kind and add an entity with properties:
+  - `property1` : john@doe.com
+  - `property2` : 343-343-3435
+- Ensure the following environment variables are set:
+   - `GOOGLE_APPLICATION_CREDENTIALS` points to authorized service account credentials file.
+   - `DLP_BUCKET_ID`  points to Google Cloud Storage bucket that contains the sample text document.
+   - `DLP_DATASTORE_KIND` points to a Datastore kind under default project.
```
Comment (lesv) [inline dlp/README.md]: Both of these are still required?

File: dlp/src/test/java/com/example/dlp/RedactIT.java (hunk 3 above)
Comment (lesv) [inline dlp/src/test/java/com/example/dlp/RedactIT.java]: doesn't look like it...

File: dlp/src/test/java/com/example/dlp/InspectIT.java (hunk 4 above)
Comment (jabubake) [inline dlp/src/test/java/com/example/dlp/InspectIT.java]: Done

**Pull request merged**